### PR TITLE
added diode and piston articles

### PR DESCRIPTION
**Pull request by nimbus98**

### CODE REVIEW

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+### What is AR (AUGMENTED REALITY)?
+
+>In layman terms, we can say that it is superimposing of computer-generated image onto the real life view of the user. It is being used almost everywhere from zoos to military training. 
+
+ ### Why AR?
+
+> AR is one of the fastest growing areas of research which will bring about a huge impact to the world, as AR is closely related to the ever fastest growing computer and technology world it will continue to grow, and we can expect a groundbreaking invention which could shake the world, like something shown bellow. 
+ 
+
+![](https://www.blockdelta.io/wp-content/uploads/2018/11/Augmented-Reality-Future.jpg)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Add a image text here

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
...ZLxNFt/XOHZvJv7x3J42ACgqofW++kVY8UDZVAqFk/0k69bwKZMNbu2oII8tIFyTFx8SnZbJPFS4XnABU1xTdg76fCO8XxDxTMwSkBixclhUnobvrFWkyUryDMEsmpdnVrqwuX7wl7QY0ZjKQpSqBzcUc8pGjX/AIhUp1sbDxOb4ps9C9mOMifKCmYglLO/YDahh7jGPCeV2jxSVilJomYUuaspQGzqAizYbgmHUkLm8TkCjkJCpiuzHKX8ojVN22e3x4qki2Yri2HlpIVOQTsDmPol4puN4ujMVAOOtBCPEjJHLhxMVvMmlIB/4Skhx3UryEIJkC55j1+0Go/SMc/2TuM4xPMv/rhDikuU+bpnI8A3BVm/piszQolyanz+JhpahuB5j5Qoqb0MZKvs2Nv0DydYf4SiUVETp0yUG5VIlhYzf1jMDl7PCHvOhjYWIVobsl8RiAn/ACzMTNRotDg+aVBwehiRlz0kBjRqRWDB8NiSKQ/Hk3TJc3jKSuJNrUHeAlUI+/MD96d4e5CVhZpclkImJUD/ALgLpIZgR5wxhZ0tVMmRZ/1CaJ/3Eh9becZGRFZdxsXE1KZZASc5UWmAkDIGdLa1Hxg3C8VLQmaVFQWUMjKAQTdlOLOExkZGWzeKYlLmb2g0uaTQAE9o1GQUWwZJJWd++qX12o3lB5WKJS1WfTUmlY3GQabFygqs6KTlBCgSSQwuGrAlzCCzmnwOsZGRtsDiro7VNJAc1LMST1t0gippByLqL1P4IyMjeTM4q6NonWcMwLaPX9yg2j18o3KxYFxd99dXftvGRkGm0copgxP5iDSrXto8NJCpqlJQxbmOUsG1vGoyM5PoyUUo8iMXiHMN4qdKyS8ucqY58zs9PD5vGRkByY941aADFDLypOYF36fMF40nGl3c2a+nfvGRkdyZvxxCKCsvvK5S/X8qDApSwdQOp/OkZGRrkwVFOwyJmVSVGzjvQj0hyXxdHvzMXnKSCGDAuwApaNRkZzZnxRl2JIxC1EBJ6CrXOp844lYaaqYUJooVPNlprXsfjGoyBk7NVR6By0kGsEeMjIOIMtjMuQWzGg3Nz2Fz3t1EBnJBs7ddfIfKsZGQS2rA6dIEoBIdmHpAFThGRkTvI7opWNVZwVCNBoyMgb2FWjGjEM4ez1baMjI5hIMqalzldtHjMwjUZDYzdC5QSZ//2Q==)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Download the image and story locally. not sure what you are displaying here.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
...3xrb36y9OkzkDtI9qPlAmS9zFdv8uI6B5A5hrQ4VmzEYa49cBLnQ117rOKOtiBzDGlxORstv0LW4uIw154z85yla49R3NYxZI5gJ67Q8xwm7tu4Rcke6dGSFWlx+0eQOYa0a49xtORw7X2HrMXnOb2eIzBxB8BlO0yag8gDfSj5mOdC90n4B4q7fWmRsiNFXBi1N1OtVtcTzEwcmecqQaETL1Y9cjmzznM9awfuGIzuc6F/T6/nYn5A1jkTp/MccXAOmWNkf27pzDzHcW75btzCZI/0JQ8rKg7KMWfzHOb9eg/9fqFde+LidBInnJnnOMRhfY4V6dGSlezvrDojTv7OqviJM+e5ZaFHSyFx40Xtla66beQ2NySud3d4g9xbSHXtgQ6iq0BZG4ihhBxEw8Cye8DeRtzPNwELdnSDL53dVni4DsT1f9kVZFjm3xnX3DC+cMDlrrsfhgQC9imuPbjPddpni2xjxPJp7RoJPc1+tLyF6g27EdK7gzKXPv0Cp4IlyHrqxXGeW4LikEB9+XNL8Rv+hb7kUXEEq7hFyFp6+QIwsA0aLcWXL2CcCuRHS/F5ziFxHchaWlyhA7hZsRG3sV4HcnRGXHN9Ll/LFyCPbMkvXwA+qFTo0TIzcVHYXhHFUdieS+KWIesEcZ4Xh+2xRu2xPRw4MmovDtsr5DwXhe0VUlyES+KwB+alxYHnljrPETlz7RX6YhV7CkG6WTI+HCgvjhUV587KpFCfE3ftgfcKCj1a6iRO5Excvs4thR4OJNS1F+OOOHHXHkgmPpTnuPRwYLauPeYnH88PT88A80H3IZOovXixPUZxW5e1Wt9+vBx0Sj1k8bpk1x6tR8fo2msPV+Y7XbTlLvjKwLLqRrv2ShOouZOhOPu6aJewZQdHNUtvjJBEsM91+/0L+6igg4eTARLwLT1abkEe595xrqL2WJl6cbqOM+GMOG2WKi4mb+Lcce1BiLv2GJF37bGifQ7KUcVNBt10SF17EdV0k/HBMp/WbRGjqL3KavRJ2Ujs0vRO23aSODrUNI3iDOrGMUrU64a5b2NEx/YSK5q2yC3cPqvk2gtSO7RDLm1+S4mGuTHuqB1ywQwaycdSMwh3yE1uxd5fb8Ui3iJXURRFCNYRSHh48zgLID12s4ordM1Ji+PVJt0uFUVRlFzzB8bIUzvArPYGAAAAAElFTkSuQmCC)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Download and render locally please

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+Object detection for using haar cascade method was created by Paul Viola and Michael Jones in their paper, "Rapid Object Detection using a Boosted Cascade of Simple Features", they have used ML to train their model and then they were able to get a very accurate model for the haar transform. 
+
+Now there are many features that can tell us whether the picture has a face or not to be exact there are 160000+ features! This is a lot of features, but the lucky thing for us is that some properties are more important than the others, for example The first feature selected focuses on the property that the region of the eyes is often darker than the region of the nose and cheeks. The second feature selected relies on the property that the eyes are darker than the bridge of the nose. These are the most prominent features!
+
+So what Paul Viola and Michael Jones did was that they group every feature into groups and extended it so when a picture fails the first group it is discarded saying that it does not contain a face.
+
+![](https://docs.opencv.org/3.4/haar.png)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Image text here

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    img = cv.imread('tom_cruise.jpg',1)
+    gray = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
+    faces = face_cascade.detectMultiScale(gray, 1.3, 5)
+    for (x,y,w,h) in faces:
+        cv.rectangle(img,(x,y),(x+w,y+h),(255,0,0),2)
+        roi_gray = gray[y:y+h, x:x+w]
+        roi_color = img[y:y+h, x:x+w]
+    cv.imshow('img',img)
+    cv.waitKey()
+    cv.destroyAllWindows()
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Enclose in back ticks.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+The above program will give us an output which will put a box around the face like this
+
+![](http://truelogic.org/wordpress/wp-content/uploads/2015/12/facedetect1.png)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Image text

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+
+Let us take a look at this program a few lines at a time.
+
+    import numpy as np
+    import cv2 as cv
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Enclose in back ticks.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+These above lines are the header folders we are importing into the program. Numpy is used for scientific calculations and cv2 is used for computer vision
+
+    face_cascad=cv.CascadeClassifier('haarcascade_frontalface_default.xml)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Enclose in back ticks.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+This line here is used to import the haar cascade filter, this is found after a lot of positive and negative picture's were used to train a model, a positive picture means it has a face on it and a negative picture means it has no face.
+
+    img = cv.imread('tom_cruise.jpg',1)
+    gray = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
+    faces = face_cascade.detectMultiScale(gray, 1.3, 5)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Enclose in back ticks.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+The first line of this code is to import a picture from the local server, the line of code used in the second line is to convert the given colour picture to a gray scale picture, and the last line is where we are appling the haar filter to the picture.
+
+    for (x,y,w,h) in faces:
+        cv.rectangle(img,(x,y),(x+w,y+h),(255,0,0),2)
+        roi_gray = gray[y:y+h, x:x+w]
+        roi_color = img[y:y+h, x:x+w]
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Enclose in back ticks.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+        
+these lines of the code we are creating a box around the face of the person in the picture.
+
+    cv.imshow('img',img)
+    cv.waitKey()
+    cv.destroyAllWindows()
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Enclose in back ticks.

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+    
+the first line is to open a window with the picture present and the box while the second states that the window should be open untill a key is pressed than after which it is destroyed.
+
+
+This is just the beginning we can add more parts to the code and some ML will help us to detect if the face is that of a male or a female and probably the age also which can be seen below
+ 
+![](https://docs.microsoft.com/en-us/azure/cognitive-services/face/images/face.detection.jpg)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Text

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+  - Military
+  - Archaeology
+  - Space research 
+  - Tourism
+
+The above mentioned area's are the largest fields in which AR is used, now I would be talking about a certain device created in this field.
+
+ This research was done with contact lenses incorporated with sensors to read the movements of the eye. This research was done in the  Optics Department at IMT-Atlantique. 
+ 
+ ![](https://i2.wp.com/skarredghost.com/wp-content/uploads/2019/05/AR_contact_lens_lentille-contact-connectee.jpg?zoom=1.5&resize=800%2C445&ssl=1)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Image Text

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+>Even before filters in snapshot or games like Pokémon-go came into the world, the military had the technology of AR and it is used for training of the soldiers in the ever changing techniques of warfare. If we had to interpret this in an easy way, it is like playing pubg in real life.
+>Have you ever seen in movies, where in a fighter jet we see all the information on the plane’s stats being displayed on the pilot’s visor, well this is not just in the movies but also in real life. This is called Tactical Augmented reality (TAR).
+Research has been going on actively at U.S. Army Research, ` Development and Engineering Command’s Communications-Electronics Research `, Development and Engineering Center (CERDEC). The biggest improvement for soldiers is the fact that they don’t have to keep looking down at their GPS to know where they are, all the information is displayed on their helmet and this improves their overall awareness.
+
+  ![](https://f1.media.brightcove.com/8/77374810001/77374810001_5453111478001_5453063309001-vs.jpg?pubId=77374810001&videoId=5453063309001)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Image text

File: blog_src/_posts/2020-01-30-AR-shaping-the-world.md [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
...kDce4bk0xu8FxWKMKkKNtdB8KrPOPwaCNjK7je+GYmTJ/lVo7PYxVTK7bbDlVQw/BMVakmwxXNAO0xuRO486sWCAgEoR5nrWGeVs7EdMlFJAPpMt97hkugxkuQBG4cQfmBTb6GuHL9YuYy4JFkZLQ63XHiI/ZT/AKgqO7b9pEuouGs6orZnfkzAEADqBJ19KL2U7dWsHhxZbC54LHOL2WcxnUZDB2HuFEW65KSir4NsPHR+A/EU4wPGlchcpEmJmsmH0o2P/pN/zH/jpafSpZXbBH/mP/HUkpUT30m4pVtoLcLmutMCJAUgg5Yka7VlmMvxmIOonUafP+dTfE+1mHxCqt205CklZOoJ38asub4VGX+JcPMnuG138b89D/vKh2+pMUl0IK9eLNJLExzMnTUan30JXjqCPL0gfCrfa4dYZVP+z77KQGUh22OoI8fnR1wGHBk8NxJJ3+1IPxk1BbqSPZDEi9hu6bXJNsgxqhHh25Qcv+Gh8Xw5Zcze0D3dw9WUeC5/iWCfOaXw7E2sPPdcMxQkAH7YHQTA1Q9T8ad38a2Itvb+p3bWZYDvfSFPI5Rbk0XXINJqmVbBuyNKzmnKAPvMdAPTnVswGFFq2EmTux/Ex1Y/15VF8E4a6t3l0QRIQabnQt8NB76mWNWyZHNlMOFY1SOMaGxrrGhsaWNOMaGxrrGhsaAOMaExpTGhMaAEsa9SGNcoAzKvV40pRrTSh1958q5vXjsK6KAOgUquV2oA9XZrleqAO16a8K9QB4mn3DbwPgPqP1FMGpVk+JfUfnQ1aJi6ZLTlMxSnthvENqLjBTfBfe9BSUzU1aJXg/aC7hiBAZeh3HoenlVr4H21ZXJ0yE7SfD5VRwKZExcWNNeVWspR9B8O7SWrsbU9v4PD3tSqk9efxFY7wRzO5+NXjhVxtNTv1pbyyTB4ItWi0J2dw+4QfE/xp1huHC2PDt06Uwwrmdz8ak7LHryq6y2V7KhGIIA1isR+kLtJ3t1sPYMWkJVyPvsNx+yNvMzWo9r7rCzdIJBFtyCDqDlOs188iiD3NvyIyLbFLzOGvJXq6KcIO12uCu0AdihXhofSjV7CibqA7Zl/eFBY2DhtrJZtKd1toPgoFFZqXcoJqhc8WoZavNSDUAeZqExpRobUAcY0NjXTQzQBxjQ2NdahtQAlmoTNSmoTUAJY16htXqCT/9k=)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-AR-shaping-the-world.md]: Download the image please

File: blog_src/_posts/2020-01-30-ekranoplan.md [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+image: '/assets/img/'
+description: 'When you think of superpowers, flight, invisibility, mind reading is what come to our mind at first. Yet, nobody thinks of Cold War Russia when we talk about superpowers. The ‘Ekranoplan’ is one such invention that was born in that super-innovative era. Is it a plane? Is it a hovercraft? Or is it something else? Read to know more'
+categories:
+- Piston
+github_username: 'Abhishek'
+---
+
+# **EKRANOPLAN**
+
+![Here's a closer look at the Ekranoplan](https://www.google.com/imgres?imgurl=https%3A%2F%2Finteng-storage.s3.amazonaws.com%2Fimages%2Fuploads%2Fsizes%2Fekranoplan_resize_md.jpg&imgrefurl=https%3A%2F%2Finterestingengineering.com%2Fa-closer-look-at-the-soviet-navys-1987-lun-class-ekranoplan&tbnid=493vh-8Bk2dxPM&vet=1&docid=rxTlTjDsu1DA5M&w=744&h=528&q=ekranoplan&source=sh%2Fx%2Fim)
```
Comment (salman-bhai) [inline blog_src/_posts/2020-01-30-ekranoplan.md]: Downlaod image. links can break anytime

**Comment by salman-bhai** [comment]
Please check this [link](https://travis-ci.org/IEEE-NITK/ieee-nitk.github.io/builds/645638035?utm_source=github_status&utm_medium=notification)

**Comment by nimbus98** [comment]
I'll make a new PR for these blogs
